packages/vite/src/node/server/middlewares/transform.ts

```diff
@@ -48,7 +54,19 @@ export function transformMiddleware(
       !req.url?.includes('vite/dist/client')
     ) {
       // missing dep pending reload, hold request until reload happens
-      server._pendingReload.then(() => res.end())
+      server._pendingReload.then(() =>
+        // If the refresh has not happened after timeout, Vite considers
+        // something unexpected has happened. In this case, Vite
+        // returns an empty response that will error.
+        setTimeout(() => {
```

Comment:
**question:** I may be wrong here, but currently the code reads like it always waits statically `5000ms` and then respond with a `408`.
What makes that it could respond earlier?
_Maybe I just misinterpret the issue and just reading the code here_

Reply:
@Shinigami92 You are absolutely right to ask.
This branch of code is reached when a dependency needs a complete refresh of the page. 
Even if the query never responds, the page is going to be refreshed and the query forgotten.

Chronologically:
- Browser asks for a dependency that is not yet optimized
- Server looks for it and starts optimization
- We reach line 57 of this file
- The server finishes optimizing the new dependency 
- Server sends a WebSocket message to the browser to refresh
- Optimization process resolves the promise on line 57/58
- Here is the change in this PR:
  - Before this PR, an empty response was given to the browser before it could refresh. This triggered the error mentioned in the ticket
  - After this PR: If the browser has not refreshed the page after 5 seconds we just drop the request

I hope this chronology clarifies the situation.